lib/src/backend.rs

```diff
@@ -180,6 +181,42 @@ pub struct CopyRecord {
     pub source_commit: CommitId,
 }
 
+/// A collection of CopyRecords.
+#[derive(Default, Debug)]
+pub struct CopyRecordMap {
+    records: Vec<CopyRecord>,
+    targets: HashMap<RepoPathBuf, usize>,
```

Comment:
originally, this was because I also had a reverse map from sources.  Now I leave it because this allows me to keep the record while removing it from the conflict list.  I strongly suspect it will continue to need this indirection.